build/generate-emoji.go

```diff
@@ -0,0 +1,134 @@
+// Copyright 2020 The Gitea Authors. All rights reserved.
+// Copyright 2015 Kenneth Shaw
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+// +build ignore
+
+package main
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"go/format"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+const (
+	gemojiURL = "https://raw.githubusercontent.com/github/gemoji/master/db/emoji.json"
+)
+
+var (
+	flagOut = flag.String("o", "modules/emoji/emoji_data.go", "out")
+)
+
+// Gemoji is a set of emoji data.
+type Gemoji []Emoji
```

Comment:
Why not import `modules/emoji`?

Reply:
modules/emoji loads the Gemoji  []struct that this file creates and so if you want to make a change here (like add a field to the struct) it will give an error about wrong number of fields in the existing data and you would need to manually edit it. 

Since this is just a generate script and not run often it seemed easier to just redefine the struct here and avoid that problem of requiring the module that it tries to modify.